src/Aspire.Dashboard/Otlp/Model/OtlpHelpers.cs

```diff
@@ -202,59 +209,126 @@ public static KeyValuePair<string, string>[] ToKeyValuePairs(this RepeatedField<
 
             var value = TruncateString(attribute.Value.GetString(), options.MaxAttributeLength);
 
-            values.Add(new KeyValuePair<string, string>(attribute.Key, value));
-
-            if (values.Count >= readLimit)
+            // If there are duplicates then last value wins.
+            var existingIndex = GetIndex(values, attribute.Key);
```

Comment:
How large can `values` get? This looks like it's O(N^2).

Reply:
There is an upper limit of 128. In the real world there probably won't be more than 10 items.

Reply:
Ok. I plan on grabbing some ETL traces from the dashboard under various loads at some point and looking for any perf bottlenecks. If this is a problem, it'll show up then.